04.ls/ls.rb

```diff
@@ -3,38 +3,76 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'etc'
+require 'date'
 
 COL_COUNT = 3
+MONTHS = 6
+
+FILE_TYPES = {
+  '01' => 'p',
+  '02' => 'c',
+  '04' => 'd',
+  '06' => 'b',
+  '10' => '-',
+  '12' => 'l',
+  '14' => 's'
+}.freeze
+
+FILE_PERMISSION_TABLES = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
+FILE_INFO_KEYS = %i[
+  file_mode
+  nlink
+  owner
+  group
+  size
+  time
+  name
+].freeze
+
+LEFT_ALIGNMENT_COLS = %i[
+  file_mode
+  name
+  owner
+  group
+].freeze
 
 def main
   success, options = parse_options
   return unless success
 
-  entry_names = find_entry_names(options)
-  max_width = entry_names.map(&:size).max
-  entry_name_table = convert_list_to_table(entry_names)
-  puts_table(entry_name_table, max_width)
+  entry_names = Dir.glob('*')
+  if options[:l]
+    entry_info_table = create_entry_info(entry_names)
```

Comment:
```suggestion
    entry_info_table = create_entry_info_table(entry_names)
```

左辺と右辺を合わせましょう
https://bootcamp.fjord.jp/pages/readable-variable-name